src/backends/github/API.js

```diff
@@ -323,42 +333,158 @@ export default class API {
+        .then(metadata => {
+          const { title, description } = options.parsedData || {};
+          const metadataFiles = get(metadata.objects, 'files', []);
+          const files = [ ...metadataFiles, ...filesList ];
+          const pr = { ...metadata.pr, head: newHead.sha };
+          const objects = {
+            entry: { path: entry.path, sha: entry.sha },
+            files: uniq(files),
+          };
+          const updatedMetadata = { ...metadata, pr, title, description, objects };
+          return this.rebasePullRequest(pr.number, branchName, contentKey, metadata, newHead);
+        });
+    }
+  }
+
+  /**
+   * Rebase a pull request onto the latest HEAD of it's target base branch
+   * (should generally be the configured backend branch). Only rebases changes
+   * in the entry file.
+   */
+  rebasePullRequest(prNumber, branchName, contentKey, metadata, head) {
+    const { path } = metadata.objects.entry;
+
+    /**
+     * Get the published branch and create new commits over it. If the pull
+     * request is up to date, no rebase will occur.
+     */
+    return this.getBranch()
+      .then(baseBranch => {
+        return this.getPullRequestCommits(prNumber, head).then(commits => {
+          return this.rebaseSingleBlobCommits(baseBranch.commit, commits, path);
+        });
+      })
+
+      /**
+       * Update metadata, then force update the pull request branch head.
+       */
+      .then(rebasedHead => {
+        const pr = { ...metadata.pr, head: rebasedHead.sha };
+        const timeStamp = new Date().toISOString();
+        const updatedMetadata = { ...metadata, pr, timeStamp };
+        return this.storeMetadata(contentKey, updatedMetadata)
+          .then(() => this.patchBranch(branchName, rebasedHead.sha, { force: true }));
       });
+  }
+
+  /**
+   * Rebase an array of commits one-by-one, starting from a given base SHA. Can
+   * accept an array of commits as received from the GitHub API. All commits are
+   * expected to change the same, single blob.
+   */
+  rebaseSingleBlobCommits(baseCommit, commits, pathToBlob) {
+    /**
+     * If the parent of the first commit already matches the target base,
+     * return commits as is.
+     */
+    if (commits.length === 0 || commits[0].parents[0].sha === baseCommit.sha) {
+      return Promise.resolve(last(commits));
     }
+
+    /**
+     * Re-create each commit over the new base, applying each to the previous,
+     * changing only the parent SHA and tree for each, but retaining all other
+     * info, such as the author/committer data.
+     */
+    const newHeadPromise = commits.reduce((lastCommitPromise, commit, idx) => {
+      return lastCommitPromise.then(newParent => {
+        /**
+         * Use `baseCommit` as the parent of the first commit, and normalize
+         * commit data to ensure it's not nested in `commit.commit`.
+         */
+        const parent = this.normalizeCommit(idx === 0 ? baseCommit : newParent);
```

Comment:
Feel free to ignore, but this ternary could be eliminated by changing the base value of the `reduce` to `Promise.resolve(baseCommit)` and then changing this line to `const parent = this.normalizeCommit(newParent);`.

Reply:
Beautiful, done.